compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEInsertLoopsForVectorization.cpp

```diff
@@ -56,11 +77,42 @@ class AMDAIEInsertLoopsForVectorizationPass
     tileSizes[numIterators - 3] = 0;
     tileSizes[numIterators - 2] = 0;
     tileSizes[numIterators - 1] = 0;
-    auto opts = linalg::LinalgTilingOptions().setTileSizes(tileSizes);
-    auto tiled = linalg::tileLinalgOp(rewriter, genericOp, opts);
-    const auto &loops = tiled.value().loops;
-    assert(!loops.empty() && "expected at least one loop here");
-    rewriter.replaceOp(genericOp, loops[0]->getResult(0));
+    return tileSizes;
+  }
+
+  /// We tile all but the innermost two dimensions currently because they form
+  /// the smallest tiled M x N dimension of the matmul.
+  static std::optional<SmallVector<int64_t>> formTileSizesForElementwise(
+      linalg::GenericOp genericOp) {
+    auto iteratorTypes = genericOp.getIteratorTypesArray();
+    auto numIterators = iteratorTypes.size();
+    assert(numIterators >= 2 && "expected at least 2 iterators here");
+    SmallVector<int64_t> tileSizes(numIterators, 1);
+    tileSizes[numIterators - 2] = 0;
+    tileSizes[numIterators - 1] = 0;
+    return tileSizes;
+  }
+
+  /// Collapse unit dims of the generic op before tiling for vectorization. Since
+  /// this is optinal we need not return failure if the collapsing cannot take
+  /// place. Eg: For <2x3x4> since there aren't any unit dimensions, it'd return
+  /// failure, hence we can simply return.
+  static void collapseUnitDims(IRRewriter &rewriter,
+                               linalg::GenericOp &genericOp,
+                               bool isTensorType) {
```

Comment:
Instead of passing this bool, I think you can derive it from the generic and simplify the function signature.